packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -102,25 +102,45 @@ class PlatformViewsService {
   ///
   /// The `id, `viewType, and `layoutDirection` parameters must not be null.
   /// If `creationParams` is non null then `creationParamsCodec` must not be null.
+  ///
+  /// Setting [surfaceController] to `true` will return a
+  /// [SurfaceAndroidViewController] which can be used with an
+  /// [AndroidViewSurface]. Setting this to `false` will return the default
+  /// [TextureAndroidViewController].
   static AndroidViewController initAndroidView({
     @required int id,
     @required String viewType,
     @required TextDirection layoutDirection,
     dynamic creationParams,
     MessageCodec<dynamic> creationParamsCodec,
     VoidCallback onFocus,
+    bool surfaceController = false,
```

Comment:
subjective: what do you think about adding a new factory method? Maybe, `initNativeAndroidView`, and `NativeAndroidViewController`.  Feel free to use a different name. :)

Reply:
That makes sense to me. I don't know if we should use the word `Native` though, since we already have a some confusion surrounding that term. E.g. `NativeAd` vs native platform view ad.